js/lib/util.js

```diff
@@ -288,25 +306,30 @@ function cropCanvas(sourceCanvas, options = {}) {
 
 	const w = sourceCanvas.width;
 	const h = sourceCanvas.height;
-	var imageData = sourceCanvas.getContext("2d").getImageData(0, 0, w, h);
+	const srcCtx = sourceCanvas.getContext("2d");
+	const offset = {
+		x: (srcCtx.origin && -srcCtx.origin.x) || 0,
+		y: (srcCtx.origin && -srcCtx.origin.y) || 0,
+	};
+	var imageData = srcCtx.getImageDataRoot(0, 0, w, h);
 	/** @type {BoundingBox} */
 	const bb = new BoundingBox();
 
-	let minx = w;
-	let maxx = -1;
-	let miny = h;
-	let maxy = -1;
+	let minx = Infinity;
+	let maxx = -Infinity;
+	let miny = Infinity;
+	let maxy = -Infinity;
```

Comment:
yeah negative infinity is one of those things i have trouble even conceptualizing lmao